compiler/src/dotty/tools/backend/jvm/CodeGen.scala

```diff
@@ -126,17 +126,16 @@ class CodeGen(val int: DottyBackendInterface, val primitives: DottyPrimitives)(
 
   // Creates a callback that will be evaluated in PostProcessor after creating a file
   private def onFileCreated(cls: ClassNode, claszSymbol: Symbol, sourceFile: util.SourceFile): AbstractFile => Unit = {
-    val (fullClassName, isLocal) = atPhase(sbtExtractDependenciesPhase) {
-      (ExtractDependencies.classNameAsString(claszSymbol), claszSymbol.isLocal)
+    val isLocal = atPhase(sbtExtractDependenciesPhase) {
```

Comment:
Could you elaborate on the commit message of this commit (local classes early). I don't really understand what it does nor why it's necessary/useful.

Reply:
sure, i can explain in the commit, but its all based upon https://github.com/sbt/zinc/commit/856d4162127927cb9a6c37a1649cc42d1871a815

Reply:
(repost as it was on the wrong thread) one thing it is actually used for in Zinc is deciding if pipelining should begin (it checks the APIs for classes that define macros, and skips pipelining if true, these API would not be available without the change)

Reply:
OK I see. Perhaps include a link to that zinc commit in the message, then.